chainer/functions/math/basic_math.py

```diff
@@ -192,8 +192,7 @@ def label(self):
     def check_type_forward(self, in_types):
         type_check.expect(in_types.size() == 2)
         type_check.expect(
-            in_types[0].dtype == numpy.float32,
-            in_types[1].dtype == numpy.float32,
+            in_types[0].dtype == in_types[1].dtype,
```

Comment:
Is it intentional that you do not impose `in_types[0].dtype.kind == 'f'`, which is added in `Div`, `PowVarVar` etc.